### PR TITLE
Adding plural entity names support

### PR DESCRIPTION
**Pull request by andersonpablosilva**
Adding plural entity names support (`datasets`, `labels`and `models`). Singular entity names are now deprecated (`dataset` and `model`).

### CODE REVIEW

File: docs/api/api_scripts/checkout_with_sample.ipynb [hunk 1]
```diff
@@ -234,8 +234,17 @@
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
    "version": "3.7.7"
+  },
```
Comment (lucianoalves) [inline docs/api/api_scripts/checkout_with_sample.ipynb:237]: Is it related pyCharm. I am not sure if it is necessary.

**Review by lucianoalves — CHANGES_REQUESTED** [review]
Why we are using dataset in there:
https://github.com/HPInc/ml-git/blob/80c8bf574fb658bad2d741d9d6ce9f24fafaace1/ml_git/repository.py#L38

**Comment by lucasncabral** [comment]
> Why we are using dataset in there:
> https://github.com/HPInc/ml-git/blob/80c8bf574fb658bad2d741d9d6ce9f24fafaace1/ml_git/repository.py#L38

The use of entities in the plural was a modification made in the CLI to facilitate the execution of commands by users.
 
Internally ml-git continues to use the 'dataset' and 'model' keys to maintain backward compatibility. Bearing in mind some points:

- metadata directories are written in .ml-git/(dataset|model|labels); 
- entities are stored inside the dataset|model|labels folder at the root of the project;
- the spec file has a key for the type of entity in the singular;
- the config file has the remote keys with the entities in the singular.

**Comment by andersonpablosilva** [comment]
Closing this PR, since this feature is on hold.

**Pull request closed without merging**